internal/application/manager.go

```diff
@@ -567,3 +595,15 @@ func (m *dataManager) processBatchedData(_ appInterfaces.AppFunctionContext, dat
 
 	return false, nil
 }
+
+func (m *dataManager) getServiceName(deviceName string) string {
+	m.recordingMutex.Lock()
+	defer m.recordingMutex.Unlock()
+
+	device := m.recordedData.Devices[deviceName]
+	if device == nil {
+		return "unknown-service"
```

Comment:
why this special name? cannot return empty string?

Reply:
Need to be non-empty name for the multi-level topic. It should never occur since all devices were loaded prior to this, so this is just CYA.

Reply:
ok